### PR TITLE
Options 1

### PR DESCRIPTION
**Pull request by itsbaska**
* Added Options cukes feature, handler, route
* Refactored Response builder to handle headers

### CODE REVIEW

File: src/main/java/Config/Routes/OPTIONSRoute.java [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import Config.Method;
+import Controller.Handler.Handler;
+
+public class OPTIONSRoute extends Route {
+	public OPTIONSRoute(Method method, String path, Handler handler) {
+		this.method = method;
+		this.path = path;
+		this.handler = handler;
+	}
+}
```
Comment (esmevane) [inline src/main/java/Config/Routes/OPTIONSRoute.java]: So, I noticed that all of these route objects have the same structure.  Is it possible you could just have one structure?  Like so:

```java
// Notice I change the interface from Route to Routed
//
public class Route extends Routed {
	public Route(Method method, String path, Handler handler) {
		this.method = method;
		this.path = path;
		this.handler = handler;
	}
}
```

Or you could have specialized custom versions, like so:

```java
public class OPTIONSRoute extends Route {
	public OPTIONSRoute(String path, Handler handler) {
		this.method = Config.Method.OPTIONS;
		this.path = path;
		this.handler = handler;
	}
}
```

File: src/main/java/Config/Routes/RouteFactory.java [hunk 2]
```diff
@@ -8,12 +8,16 @@
...
+      switch (method) {
+        case GET:
+          return new GETRoute(method, path, handler);
+        case POST:
+          return new POSTRoute(method, path, handler);
+        case OPTIONS:
+          return new OPTIONSRoute(method, path, handler);
+        default:
+          throw new Error();
+      }
```
Comment (esmevane) [inline src/main/java/Config/Routes/RouteFactory.java]: Re: my above feedback, if you go with a generic structure, here you could reduce the switch statement to this:

```java
return new Route(method, path, handler);
```

And just trust your `Method.toMethod` approach with the error handling.

File: src/main/java/Controller/Handler/OPTIONSHandler.java [hunk 3]
```diff
@@ -0,0 +1,20 @@
+package Controller.Handler;
+import Config.Method;
+import Request.Request;
+import Response.Response;
+
+public class OPTIONSHandler extends Handler{
+  public OPTIONSHandler() {
+    this.method = Method.POST;
```
Comment (esmevane) [inline src/main/java/Controller/Handler/OPTIONSHandler.java]: Did you mean to set this to `POST`?

Comment (itsbaska) [reply src/main/java/Controller/Handler/OPTIONSHandler.java]: Oh whoooops...

**Pull request merged**